google-mail-workitem/src/main/java/org/jbpm/workitem/google/mail/SendMailWorkitemHandler.java

```diff
@@ -0,0 +1,196 @@
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "GoogleSendMailDefinitions.wid", name = "GoogleSendMail",
+        displayName = "GoogleSendMail",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.google.mail.SendMailWorkitemHandler()",
+        parameters = {
+                @WidParameter(name = "AppName"),
+                @WidParameter(name = "ClientSecret"),
+                @WidParameter(name = "To"),
+                @WidParameter(name = "From"),
+                @WidParameter(name = "Subject"),
+                @WidParameter(name = "BodyText"),
+                @WidParameter(name = "Attachment")
+        },
+        results = {
+                @WidResult(name = "Message")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "com.google.api-client", artifact = "google-api-client", version = "1.23.0"),
+                @WidMavenDepends(group = "com.google.oauth-client", artifact = "google-oauth-client-jetty", version = "1.23.0"),
+                @WidMavenDepends(group = "com.google.apis", artifact = "google-api-services-gmail", version = "v1-rev72-1.23.0")
+        })
+public class SendMailWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(SendMailWorkitemHandler.class);
+    private static final String RESULTS_MESSAGE = "Message";
+
+    private GoogleMailAuth auth = new GoogleMailAuth();
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        Map<String, Object> results = new HashMap<String, Object>();
+        String paramAppName = (String) workItem.getParameter("AppName");
+        String paramClientSecretJSON = (String) workItem.getParameter("ClientSecret");
+        String paramTo = (String) workItem.getParameter("To");
+        String paramFrom = (String) workItem.getParameter("From");
+        String paramSubject = (String) workItem.getParameter("Subject");
+        String paramBodyText = (String) workItem.getParameter("BodyText");
+        Document paramAttachment = (Document) workItem.getParameter("Attachment");
+
+        try {
+            Gmail gmailService = auth.getGmailService(paramAppName,
+                                                      paramClientSecretJSON);
+            Message outEmailMessage = sendMessage(gmailService,
+                                              paramTo,
+                                              paramFrom,
+                                              paramSubject,
+                                              paramBodyText,
+                                              paramAttachment);
+
+            results.put(RESULTS_MESSAGE,
+                        outEmailMessage);
```

Comment:
why do we need the message to be returned? Asking as it might come with some data that won't be serialisable and thus will fail the process instance serialisation.